src/main/scala/kamon/datadog/DatadogAgentReporter.scala

```diff
@@ -100,25 +100,111 @@ class DatadogAgentReporter extends MetricReporter {
     case _                                                            => value.toDouble
   }
 
-  private def readConfiguration(config: Config): Configuration = {
+}
+
+object DatadogAgentReporter {
+
+  private val logger = LoggerFactory.getLogger(classOf[DatadogAgentReporter])
+
+  trait MeasurementFormatter {
```

Comment:
May I ask why introduce this trait? I see the `PacketBuffer` useful for testing purposes but here we don't have real room for flexibility since the Datadog spec is fixed. After introducing the EnvironmentTagBuilder it also seems like most of the implementation could go away as well.

Reply:
So, you're right. The decision here is rather questionable, but it comes from the fact that I'm actually using this for just a statsD with tags rather than sending it to datadog explicitly (on that note, it might make sense to merge this with the statsD lib. We could use this measurementformatter to select between statsD vs dogstatsd).

The benefit though is that at some point, when I upgrade to kamon 1.0, all of the graphs we wrote against the old format are going to break. I was thinking about how I could shim a compatibility layer into the reporter to report both or something without having to write a separate reporter to do the translation of the new formats to the old.

But I'm happy to drop this configurability out.